dora/core/common/src/main/java/alluxio/membership/AlluxioEtcdClient.java

```diff
@@ -127,36 +124,41 @@ public static AlluxioEtcdClient getInstance(AlluxioConfiguration conf) {
     return sAlluxioEtcdClient;
   }
 
-  /**
-   * Create jetcd grpc client no forcing.
-   */
-  public void connect() {
-    connect(false);
+  @FunctionalInterface
+  protected interface EtcdUtilCallable<V> {
+    /**
+     * The task where logics to communicate with etcd happens.
+     *
+     * @return Etcd gRPC call result
+     * @throws Exception
+     */
+    V call() throws Exception;
   }
 
-  /**
-   * Create jetcd grpc client and force(or not) connection.
-   * @param force
-   */
-  public void connect(boolean force) {
-    if (mConnected.get() && !force) {
-      return;
-    }
-    mConnected.set(false);
-    // create client using endpoints
-    Client client = Client.builder().endpoints(mEndpoints)
-        .build();
-    if (mConnected.compareAndSet(false, true)) {
-      mClient = client;
+  private <V> V retryInternal(String description, RetryPolicy retryPolicy,
+                              EtcdUtilCallable<V> etcdCallable) throws IOException {
+    Exception ex = null;
+    // TODO(lucy) Currently retry on all sorts of exception and report the last exception,
+    // As jetcd exception often hides underneath CompletableFuture.get(), find
+    // more info later on distinguishing different possible exceptions.
+    while (retryPolicy.attempt()) {
+      try {
+        return etcdCallable.call();
+      } catch (Exception e) {
+        // TODO(lucy) check in future if io.etcd.jetcd.common.exception.EtcdException
+        // needs to be handled for detection of underlying jetcd client connection
+        // getting into possible stale states or needs reconnection, for now, reuse single
+        // jetcd client handle for entire lifecycle of AlluxioEtcdClient.
+        LOG.warn("Failed to {} (attempt {}): {}", description, retryPolicy.getAttemptCount(),
```

Comment:
https://github.com/etcd-io/jetcd/blob/main/jetcd-common/src/main/java/io/etcd/jetcd/common/exception/EtcdExceptionFactory.java#L67
checkout this and it seems like they have a similar exception system that uses GRPC status as the indicator, so we can retry on certain exceptions like how we handle grpc call.